presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -318,7 +318,7 @@ public int getQueriesQueuedOnInternal()
     private static int getQueriesQueuedOnInternal(InternalResourceGroup resourceGroup)
     {
         if (resourceGroup.subGroups().isEmpty()) {
-            return Math.min(resourceGroup.getQueuedQueries(), resourceGroup.getMaxRunningQueries() - resourceGroup.getRunningQueries());
+            return Math.min(resourceGroup.getQueuedQueries(), resourceGroup.getHardConcurrencyLimit() - resourceGroup.getRunningQueries());
```

Comment:
In [this](https://github.com/prestodb/presto/pull/9105#discussion_r143825387) comment you were saying that this should be the soft limit.

Reply:
I changed this to soft in the commit where I added soft.